server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -5266,6 +5270,16 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
             throw new InvalidParameterValueException("Unable to use template " + templateId);
         }
 
+        ServiceOfferingJoinVO svcOffering = serviceOfferingJoinDao.findById(serviceOfferingId);
+
```

Comment:
@Pearl1594 is it possible to move all the checks after `template.isDeployAsIs()` to a separate if stmt (instead calling "template.isDeployAsIs()" multiple times)

Reply:
@sureshanaparti done

Reply:
checked, looks good now.